pkg/storegateway/indexheader/reader_pool.go

```diff
@@ -51,21 +55,83 @@ type ReaderPool struct {
 	lazyReaders   map[*LazyBinaryReader]struct{}
 }
 
-// NewReaderPool makes a new ReaderPool and starts a background task for unloading idle Readers if enabled.
-func NewReaderPool(logger log.Logger, lazyReaderEnabled bool, lazyReaderIdleTimeout time.Duration, metrics *ReaderPoolMetrics) *ReaderPool {
+type HeadersLazyLoaded struct {
+	Path   string
+	UserID string
+}
+
+type HeadersLazyLoadedTrackerState struct {
```

Comment:
Mirroring Dimitar's suggestion for the struct above, how about calling this something like `LazyLoadedHeadersSnapshot`?